packages/rpc-core/src/rpc-methods/getTransaction.ts

```diff
@@ -0,0 +1,254 @@
+    };
+}>;
+
+type InnerInstructions<InstructionType> = Readonly<{
+    index: number;
+    instructions: readonly InstructionType[];
+}>;
+
+type TransactionMetaInnerInstructionsNotParsed = Readonly<{
+    innerInstructions: readonly InnerInstructions<TransactionInstruction>[];
+}>;
+
+type TransactionMetaInnerInstructionsParsed = Readonly<{
+    innerInstructions: readonly InnerInstructions<TransactionInstructionNoParser | ParsedTransactionInstruction>[];
+}>;
+
+type TransactionAddressTableLookups = Readonly<{
+    message: Readonly<{
+        addressTableLookups?: readonly AddressTableLookup[] | null;
+    }>;
+}>;
+
+export interface GetTransactionApi {
+    /**
+     * Returns transaction details for a confirmed transaction
+     */
+    getTransaction<TMaxSupportedTransactionVersion extends TransactionVersion | void = void>(
+        address: Base58EncodedAddress,
+        config: GetTransactionCommonConfig<TMaxSupportedTransactionVersion> &
+            Readonly<{
+                encoding: 'jsonParsed';
+            }>
+    ): GetTransactionApiResponseBase &
+        (TMaxSupportedTransactionVersion extends void ? Record<string, never> : { version: TransactionVersion }) & {
+            meta: TransactionMetaBase & TransactionMetaInnerInstructionsParsed;
+            transaction: TransactionJsonParsed & TransactionAddressTableLookups;
+        };
+    getTransaction<TMaxSupportedTransactionVersion extends TransactionVersion | void = void>(
+        address: Base58EncodedAddress,
+        config: GetTransactionCommonConfig<TMaxSupportedTransactionVersion> &
+            Readonly<{
+                encoding: 'base64';
+            }>
+    ): GetTransactionApiResponseBase &
+        (TMaxSupportedTransactionVersion extends void ? Record<string, never> : { version: TransactionVersion }) & {
+            meta: TransactionMetaBase & TransactionMetaLoadedAddresses & TransactionMetaInnerInstructionsNotParsed;
+            transaction: Base64EncodedDataResponse;
+        };
+    getTransaction<TMaxSupportedTransactionVersion extends TransactionVersion | void = void>(
+        address: Base58EncodedAddress,
+        config: GetTransactionCommonConfig<TMaxSupportedTransactionVersion> &
+            Readonly<{
+                encoding: 'base58';
+            }>
+    ): GetTransactionApiResponseBase &
+        (TMaxSupportedTransactionVersion extends void ? Record<string, never> : { version: TransactionVersion }) & {
+            meta: TransactionMetaBase & TransactionMetaLoadedAddresses & TransactionMetaInnerInstructionsNotParsed;
+            transaction: Base58EncodedDataResponse;
+        };
+    getTransaction<TMaxSupportedTransactionVersion extends TransactionVersion | void = void>(
+        address: Base58EncodedAddress,
+        config: GetTransactionCommonConfig<TMaxSupportedTransactionVersion> &
+            Readonly<{
+                encoding?: 'json';
+            }>
+    ): GetTransactionApiResponseBase &
```

Comment:
All of these return types need to incorporate the possibility that the return is `null`.